mitype/app.py

```diff
@@ -198,10 +198,18 @@ def key_printer(self, win, key):
         # Check for space
         elif key == " ":
             self.check_word()
-
-        # Check for any other typable characters
+        
+         # Check for any other typable characters
         elif mitype.keycheck.is_valid_initial_key(key):
-            self.appendkey(key)
+
+            # Check if input is a special character
+            if sys.version_info[0] < 3:
+                if(not(mitype.keycheck.is_ignored(key))):
+                    self.appendkey(key)
+
+            else:
+                if(not(isinstance(key,int))):
+                    self.appendkey(key)
```

Comment:
We gotta change the way this code is structured.
Put this block of code inside the `keycheck.is_ignored` method.